rt/libuv/network.mbt

```diff
@@ -0,0 +1,433 @@
+  udp : @uv.Udp
+  rt : LibuvRuntime
+  local : @net.SockAddr?
+}
+
+///|
+impl @net.Socket for UvDatagramSocket with accept(self, timeout) {
+  let buffer = @buffer.Buffer::new()
+  self.accept(buffer, timeout).map(fn(pair) { pair.0 })
+}
+
+///|
+impl @net.Socket for UvDatagramSocket with close(self) {
+  guard !self.udp.is_closing()
+  uvrt_op_suspend((resolve, _, _) => {
+    self.udp.close(() => resolve(()))
+    () => ()
+  })
+}
+
+///|
+impl @net.Socket for UvDatagramSocket with is_closed(self) {
+  self.udp.is_closing()
+}
+
+///|
+impl @net.DatagramSocket for UvDatagramSocket with accept(self, buffer, timeout) {
+  let _ = timeout
+  let recv_buffer = FixedArray::make(65536, 0)
+  uvrt_op_suspend((resolve, reject, _) => {
+    self.udp.recv_start(
+      (_, _) => recv_buffer.unsafe_reinterpret_as_bytes(),
+      (_, nread, sockaddr, _) => {
+        self.udp.recv_stop()
+        if nread > 0 {
+          buffer.write_bytes(recv_buffer.unsafe_reinterpret_as_bytes()[0:nread])
+          let addr = uv_to_sockaddr(sockaddr)
+          resolve((addr, nread))
+        } else {
+          reject(@net.NetworkError::ConnClose(msg="Connection closed"))
+        }
+      },
+      (_, err) => {
+        self.udp.recv_stop()
+        reject(@net.NetworkError::DeviceFailure(msg="Receive failed: \{err}"))
+      },
+    ) catch {
+      err => reject(@net.NetworkError::DeviceFailure(msg="Failed to start receive: \{err}"))
+    }
+    () => {
+      defer reject(@rt.CoroutineError::Cancelled)
+      self.udp.recv_stop()
+    }
+  })
+}
+
+///|
+impl @net.DatagramSocket for UvDatagramSocket with send(self, dst, buffer, opt) {
+  let _ = opt
+  let data = buffer.to_bytes()
+  uvrt_op_suspend((resolve, reject, _) => {
+    let handle = self.udp.send(
+      [data],
+      sockaddr_to_uv(dst),
+      () => resolve(()),
+      err => reject(@net.NetworkError::DeviceFailure(msg="Send failed: \{err}")),
+    )
+    () => {
+      defer reject(@rt.CoroutineError::Cancelled)
+      handle.cancel()
+    }
+  })
+}
+
+///|
+struct UvDatagramConnection {
+  udp : @uv.Udp
+  rt : LibuvRuntime
+  sockaddr : @net.SockAddr
+  buffer : &@buffer.Buffer
+}
+
+///|
+impl @net.Connection for UvDatagramConnection with local_addr(self) {
+  @net.SockAddr::{ inet: @net.InetAddr::IPv4(host=0), port: 0 }
+}
+
+///|
+impl @net.Connection for UvDatagramConnection with remote_addr(self) {
+  self.sockaddr
+}
+
+///|
+impl @net.Connection for UvDatagramConnection with flush(self) {
+  // UDP doesn't need flushing
+  ()
+}
+
+///|
+impl @io.Channel for UvDatagramConnection with read(self, buffer, offset, length) {
+  let bytes_read = self.buffer.read_bytes(buffer.unsafe_reinterpret_as_bytes()[offset:offset + length])
+  bytes_read
+}
+
+///|
+impl @io.Channel for UvDatagramConnection with write(self, buffer, offset, length) {
+  let stream_buffer = @buffer.Buffer::new()
+  stream_buffer.write_bytes(buffer.unsafe_reinterpret_as_bytes()[offset:offset + length])
```

Comment:
No we dont have this method. Check the definition of Buffer

there should be a `transfer` method you can use by slicing out of the parameter buffer